### PR TITLE
Make “read” to be datetime.

### PR DESCRIPTION
**Pull request by drets**

### CODE REVIEW

File: heutagogy/schema.sql [hunk 1]
```diff
@@ -4,5 +4,5 @@ CREATE TABLE bookmarks (
   timestamp text,
   url text not null,
   title text,
-  read bool
+  read text
```
Comment (rasendubi) [inline heutagogy/schema.sql:7]: We are using SQLAlchemy, so we don't need this schema file now.

File: tests.py [hunk 2]
```diff
@@ -144,13 +144,14 @@ def test_new_bookmark_post_is_unread(self):
...
         self.assertEqual(HTTPStatus.CREATED, res.status_code)
-        self.assertFalse(result['read'])
+        self.assertIsNone(result['read'])
 
     @single_user
     def test_new_bookmark_read_is_unread(self):
         bookmark = {
             'url': 'https://github.com/',
         }
+        read = '2016-11-06T01:31:15'
```
Comment (rasendubi) [inline tests.py:154]: Inline this (or move close to the usage)

File: tests.py [hunk 3]
```diff
@@ -162,18 +163,19 @@ def test_new_bookmark_read_is_unread(self):
         res = self.app.get(
             '/api/v1/bookmarks/{}'.format(bookmark_id),
             content_type='application/json',
-            data=json.dumps({'read': True}),
+            data=json.dumps({'read': read}),
```
Comment (rasendubi) [inline tests.py:166]: oh, that's a wrong test. `GET /api/v1/bookmarks/{}` should not have a payload.

**Comment by rasendubi** [comment]
Maybe, add a feature to insert current timestamp if the user doesn't pass one. (e.g., `{ "read": true }`)

File: heutagogy/schema.sql (hunk 1 above)
Comment (drets) [reply heutagogy/schema.sql:7]: Thanks for clarifying. I wasn't sure why we need this file too.

File: tests.py (hunk 3 above)
Comment (drets) [reply tests.py:166]: Would you create an issue for that?

**Comment by drets** [comment]
> Maybe, add a feature to insert current timestamp if the user doesn't pass one. (e.g., { "read": true })

Now it's working like that:
• if user sent a non-empty `read` flag we assume that it's a timestamp
• if user didn't send a non-empty `read` flag we set it to `None`

Do you want instead of `None` add current timestamp; check that timestamp which user sent is valid timestamp?

File: tests.py (hunk 3 above)
Comment (drets) [reply tests.py:166]: It seems it's going to be fixed by this patch: https://github.com/heutagogy/heutagogy-backend/pull/55

**Comment by rasendubi** [comment]
I want next semantics:
- if `read` is absent, we assume default value (`None` for new bookmarks, unchanged for updating)
- if `read` is present, but `null` (or `false`), the bookmark is unread.
- if `read` is present and `true`, we mark it as read and use current timestamp.
- otherwise, parse `read` as ISO 8601 timestamp.

**Pull request merged**